packages/data-broker/src/services/dhis/DhisService.js

```diff
@@ -216,6 +216,17 @@ export class DhisService extends Service {
     return dataSources.filter(({ code }) => allowedDataSourceCodes.includes(code));
   };
 
+  validateProgramCodesExist = async programCodes => {
+    const validateProgramCode = async programCode => {
+      const programDataSource = await this.models.dataSource.findOne({
+        code: programCode,
+        type: this.dataSourceTypes.DATA_GROUP,
+      });
+      if (!programDataSource) throw new Error(`Program not found: ${programCode}`);
+    };
+    return Promise.all(programCodes.map(programCode => validateProgramCode(programCode)));
```

Comment:
It would be more efficient to fetch data sources using all `programCodes` with a single `find` query, then check whether any one is missing in the results